alacritty_terminal/src/term/mod.rs

```diff
@@ -1356,20 +1351,32 @@ impl Term {
                 parser.reset();
             }
 
+            let above_viewport = point.line > last_line.0 + display_offset;
+
             // Advance parser
             match parser.advance(cell.c) {
                 ParserState::Url(length) => {
-                    self.urls.push(Url::new(point, length + extra_url_len, num_cols))
+                    current_url_len = 0;
+
+                    self.urls.push(Url::new(point, length + extra_url_len, num_cols));
                 },
                 ParserState::NoUrl => {
                     extra_url_len = 0;
+                    current_url_len = 0;
 
                     // Stop searching for URLs once the viewport has been left without active URL
-                    if point.line > last_line.0 + display_offset {
+                    if above_viewport {
                         break;
                     }
                 },
-                _ => (),
+                ParserState::MaybeUrl => {
+                    current_url_len += 1;
```

Comment:
Just a note since I've caught this change, this does not accurately track the URL length. When characters like brackets or quotes are included in the URL, the URL will be truncated when there is no matching opening quote/bracket.

Reply:
It doesn't need to be super accurate; this is just used to make sure the URL isn't too long above the viewport